syntax = "proto3";

package bibliophage;

// Service definitions - the API endpoints
// RPC method: takes Request, returns Response
service LoadingService {
  rpc LoadPDF(PdfLoadRequest) returns (PdfLoadResponse);
}

// Request messages - what the client sends
message PdfLoadRequest {
  string pdf_origin_path = 1; // Filesystem path to the PDF
  string pdf_name = 2; // Human-readable name (e.g., "Monster Manual")
  string pdf_system = 3; // Which RPG system
  string pdf_type = 4; // Type of publication
  int32 pdf_page_count = 5; // Number of pages (optional, can be 0)

  // Chunking parameters
  optional int32 chunk_size = 6; // Size of text chunks (default: 600)
  optional int32 chunk_overlap = 7; // Overlap between chunks (default: 50)

  // PDF file data
  bytes file_data = 8;
}

// Response messages - what the server sends back
message PdfLoadResponse {
  bool success = 1; // Did it work?
  string message = 2; // Human-readable status/error message
  int32 chunks_created = 3; // How many chunks were stored
  string document_id = 4; // Unique ID for this ingested document
}
